packages/boot/tools/supports.ts

```diff
@@ -449,34 +447,13 @@ export const makeSwingsetTestKit = async (
           case 'startChannelOpenInit':
             pushInbound(BridgeId.DIBC, icaMocks.channelOpenAck(obj));
             return undefined;
-          case 'sendPacket':
-            switch (obj.packet.data) {
-              case protoMsgMocks.delegate.msg: {
-                return ackLater(obj, protoMsgMocks.delegate.ack);
-              }
-              case protoMsgMocks.delegateWithOpts.msg: {
-                return ackLater(obj, protoMsgMocks.delegateWithOpts.ack);
-              }
-              case protoMsgMocks.queryBalance.msg: {
-                return ackLater(obj, protoMsgMocks.queryBalance.ack);
-              }
-              case protoMsgMocks.queryUnknownPath.msg: {
-                return ackLater(obj, protoMsgMocks.queryUnknownPath.ack);
-              }
-              case protoMsgMocks.queryBalanceMulti.msg: {
-                return ackLater(obj, protoMsgMocks.queryBalanceMulti.ack);
-              }
-              case protoMsgMocks.queryBalanceUnknownDenom.msg: {
-                return ackLater(
-                  obj,
-                  protoMsgMocks.queryBalanceUnknownDenom.ack,
-                );
-              }
-              default: {
-                // An error that would be triggered before reception on another chain
-                return ackImmediately(obj, protoMsgMocks.error.ack);
-              }
+          case 'sendPacket': {
+            if (protoMsgMockMap[obj.packet.data]) {
+              return ackLater(obj, protoMsgMockMap[obj.packet.data]);
             }
+            // An error that would be triggered before reception on another chain
+            return ackImmediately(obj, protoMsgMocks.error.ack);
```

Comment:
> // An error that would be triggered before reception on another chain

Thinking on this more, I suspect the relayer could also be delayed. Any harm in making this `ackLater`?

Reply:
Thinking on this more, I also don't have a strong stance. It seems the main benefit of `ackLater` is to allow us to control timing in the test context. `ackImmediately` seems like a sensible default, as someone can always choose to supply a mock to the bridge that will use `ackLater`. 

Nonetheless, sharing some thoughts/findings to your question - 

> 1. a sendPacket message could have an immediate failure before leaving the local node (i.e. just over the bridge) or after a reply from a relayer.

Correct, the [sendPacket handler](https://github.com/Agoric/agoric-sdk/blob/ca60e8daa50a095b5ee339c042cfd7f467566179/golang/cosmos/x/vibc/types/receiver.go#L124) can fail via [ReceiveSendPacket](https://github.com/Agoric/agoric-sdk/blob/80e450a9afe0441406fa3e020d94ef0af6366851/golang/cosmos/x/vibc/keeper/keeper.go#L113-L129) before leaving agoric golang and going out to a relayer.

It _seems_ this might only fail if an invalid `sourcePort` and `sourceChannel` are provided. I'm not sure when `ibc.js` or `network.js` would allow this  since `.downcall()` is heavily guarded with ocaps + closures - the only thing coming to mind is bad state where a channel is closed but ibc/network are unaware.